File: StocksProcessing.Analyzer/src/main/java/org/example/models/messages/CurrencyRatesStatisticsMessage.java
package org.example.models.messages;

import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonProperty;
import org.example.models.Currency;

public class CurrencyRatesStatisticsMessage {
    @JsonProperty("stock_exchange_name")
    public String stockExchangeName;

    @JsonProperty("currency")
    public Currency currency;

    @JsonProperty("rate_in_rubles")
    public Double rateInRubles;

    public CurrencyRatesStatisticsMessage(
            String stockExchangeName,
            Currency currency,
            Double rateInRubles
    ) {
        this.stockExchangeName = stockExchangeName;
        this.currency = currency;
        this.rateInRubles = rateInRubles;
    }
}

File: StocksProcessing.Streamer/Models/Messages/CurrencyRatesStatisticsMessage.cs
using System.Text.Json.Serialization;
using StocksProcessing.Streamer.Models.Enums;

namespace StocksProcessing.Streamer.Models.Messages;

public sealed record CurrencyRatesStatisticsMessage
{
    public required string StockExchangeName { get; init; }

    [JsonConverter(typeof(JsonStringEnumConverter))]
    public required Currency Currency { get; init; }

    public required decimal RateInRubles { get; init; }
}